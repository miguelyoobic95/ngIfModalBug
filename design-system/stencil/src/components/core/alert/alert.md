---
name: Alert
category: Core Components
---

Alerts can be used with `<yoo-alert>` elements.

## Attributes
|Name|Type|Default|Description|
|---|---|---|---|
|`text`|string|   |the text inside the alert|
|`heading`|string|    |the heading of the alert|
|`icon`|string|   |the icon in the alert|
|`closeable`|boolean|   |the alert is closable (default = false)|

## Variations

```yoo-alert-variations.html
    <yoo-alert class="accent toast" heading="Heading" text="text"></yoo-alert>
    <br>
    <yoo-alert class="accent toast" heading="Closeable" text="accent toast and closeable" closeable="true"></yoo-alert>
    <br>
    <yoo-alert class="accent toast" heading="Icon" text="accent toast with icon" icon="yo-check-tick" closeable="true"></yoo-alert>
```

```yoo-alert-variations.css hidden
    yoo-alert {
        margin-bottom:5px;
    }
```

## CSS Classes

|Type|Name|Description|
|---|---|---|
|Toast|`toast`|alert of type toast, placed on the top right corner of the screen|
|Embedded|`embedded`|alert of type embedded, placed inside forms|
|Stripe|`stripe`|alert of type stripe, placed below the last navigation element, either the header or the navigation bar|
|Card|`card`|alert of type card|
|Round|`round`|round alert|
|Accent|`accent`|alert accent|
|Danger|`danger`|alert danger|
|Success|`success`|alert success|
|Info|`info` |alert info|
|Warning|`warning`|alert warning|
|Accent Gradient|`accent-gradient`|alert accent-gradient|
|Accent Gradient Card|`accent-gradient-card`|alert accent-gradient to be used with card class|
|Danger Gradient|`danger-gradient`|alert danger-gradient|
|Danger Gradient Card|`danger-gradient-card`|alert danger-gradient to be used with card class|
|Success Gradient|`success-gradient`|alert success-gradient|
|Success Gradient Card|`success-gradient-card`|alert success-gradient to be used with card class|
|Info Gradient|`info-gradient` |alert info-gradient|
|Info Gradient Card|`info-gradient-card` |alert info-gradient to be used with card class|
|Warning Gradient|`warning-gradient`|alert warning-gradient|
|Warning Gradient Card|`warning-gradient-card`|alert warning-gradient to be used with card class|

```yoo-alert-types.html
    <yoo-alert class="accent toast" heading="Toast" text="accent toast"></yoo-alert>
    <br>
    <yoo-alert class="accent round toast" heading="Toast" text="accent round toast"></yoo-alert>
    <br>
    <yoo-alert class="accent embedded" heading="Embedded" text="accent embedded"></yoo-alert>
    <br>
    <yoo-alert class="accent round embedded" heading="Embedded" text="accent round embedded"></yoo-alert>
    <br>
    <yoo-alert class="accent stripe" heading="Stripe" text="accent stripe"></yoo-alert>
    <br>
    <yoo-alert class="accent card" heading="Card" text="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
" closeable="true" icon="yo-check-tick"></yoo-alert>
    <br>
     <yoo-alert class="accent-gradient-card card" heading="Card" text="accent gradient card" closeable="true" icon="yo-check-tick"></yoo-alert>
    <br>
    <yoo-alert class="danger-gradient-card card" heading="Card" text="danger gradient card" closeable="true" icon="yo-check-tick"></yoo-alert>
    <br>
    <yoo-alert class="success-gradient-card card" heading="Card" text="success gradient card" closeable="true" icon="yo-check-tick"></yoo-alert>
    <br>
    <yoo-alert class="info-gradient-card card" heading="Card" text="info gradient card" closeable="true" icon="yo-check-tick"></yoo-alert>
    <br>
    <yoo-alert class="warning-gradient-card card" heading="Card" text="warning gradient card" closeable="true" icon="yo-check-tick"></yoo-alert>
    <br>
    <yoo-alert class="accent embedded" heading="Class" text="accent embedded"></yoo-alert>
    <br>
    <yoo-alert class="danger embedded" heading="Class" text="danger embedded"></yoo-alert>
    <br>
    <yoo-alert class="success embedded" heading="Class" text="success embedded"></yoo-alert>
    <br>
    <yoo-alert class="info embedded" heading="Class" text="info embedded"></yoo-alert>
    <br>
    <yoo-alert class="warning embedded" heading="Class" text="warning embedded"></yoo-alert>
    <br>
    <yoo-alert class="accent-gradient embedded" heading="Class" text="accent-gradient embedded"></yoo-alert>
    <br>
    <yoo-alert class="danger-gradient embedded" heading="Class" text="danger-gradient embedded"></yoo-alert>
    <br>
    <yoo-alert class="success-gradient embedded" heading="Class" text="success-gradient embedded"></yoo-alert>
    <br>
    <yoo-alert class="info-gradient embedded" heading="Class" text="gradient-info embedded"></yoo-alert>
    <br>
    <yoo-alert class="warning-gradient embedded" heading="Class" text="warning-gradient embedded"></yoo-alert>
```

```yoo-alert-types.css hidden
    yoo-alert {
        margin-bottom:5px;
    }
```

## Events

|Attr|Description|
|---|---|---|
|`alertClosed`|event emitted when the alert is closed|
|`alertActionSelected`|event emitted when the action text is pressed|