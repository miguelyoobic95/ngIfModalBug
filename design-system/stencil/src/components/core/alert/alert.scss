

:host {
    display: block;
    &.closed {
        opacity: 0;
    }
    .container {
        display: inline-flex;
        align-items: center;
        justify-content: space-between;
        padding: 0.75rem 1rem;
        transition: all 0.3s;
        width:100%;
        background: red;
        &.closed {
            opacity: 0;
        }
        .link {
            color: #ffffff;
            font-weight: 100;
            text-decoration: underline;
            cursor: pointer;
            opacity: 0.5;
            font-size: 0.875rem;            
        }
        .inner-container {
            display: flex;
            justify-content: center;
            .icon {
                font-family: 'yoobicons';
                margin-right: 0.5rem;
                color: #ffffff;
            }
            .text-container {
                display: flex;
                justify-content: center;
                color: #ffffff;
                .value {
                    line-height: 1rem;
                    margin-bottom: 0.25rem;
                    font-size: 0.875rem;
                }
                .heading {
                    align-self: baseline;
                    font-weight: bold;
                    font-size: 0.875rem;
                }
            }
        }
        .close-container {
            display: flex;
            justify-content: flex-end;
            color: #ffffff;
            .close {
                cursor: pointer;
                font-family: 'yoobicons';
                margin-left: 1rem;
                justify-content: flex-end;
            }
        }
    }
}

:host(.round) {
    .container {
        border-radius: 0.5rem;
    }
}

