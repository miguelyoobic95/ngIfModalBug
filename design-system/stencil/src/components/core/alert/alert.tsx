import { Component, Prop, State, Event, EventEmitter, Element } from '@stencil/core';
import { setAnimation } from '../../../utils/anim';

@Component({
    tag: 'yoo-alert',
    styleUrl: 'alert.scss',
    scoped: true
})
export class YooAlertComponent {

    @Prop() text: string;
    @Prop() heading: string;
    @Prop() icon: string;
    @Prop() closeable: boolean;
    @Prop() animationName: string;
    @Prop() link: string;

    @Event() alertClosed: EventEmitter<boolean>;
    @Event() alertActionSelected: EventEmitter<boolean>;

    @State() closed: boolean;

     @Element() host: HTMLStencilElement;

    componentDidLoad() {
        setAnimation(this.animationName, this.host, {open: true});
    }

    onActionTextClick(): void {
        this.alertActionSelected.emit(true);
    }

    onDismissButtonClick(): void {
        this.alertClosed.emit(true);
        this.closed = true;
    }

    render(): JSX.Element {
        return (
            <div class={this.closed ? 'container closed' : 'container'}>
                <div class="link" onClick={() => this.onActionTextClick()}>
                    {this.link}
                </div>
                <div class = "inner-container">
                    {this.icon ? <span class="icon"><i class={this.icon}></i></span> : null}
                    <div class="text-container">
                        {this.heading ? <span class="heading">{this.heading}</span> : null}
                        <span class="value">{this.text}</span>
                    </div>
                </div>
                <div class="close-container">
                    {this.closeable ? <span class="close" onClick={() => this.onDismissButtonClick()}> <i class="yo-close"></i></span> : null}
                </div>

            </div>
        );
    }
}
