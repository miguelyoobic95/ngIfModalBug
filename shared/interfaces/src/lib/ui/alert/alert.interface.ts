export interface IAlertEntry {
    heading?: string;
    icon?: string;
    text?: string;
    closeable?: boolean;
    cssClass?: string;
    animationName?: string;
    withYooCtrl?: boolean;
}
