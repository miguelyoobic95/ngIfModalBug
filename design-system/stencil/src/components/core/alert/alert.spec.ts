import { TestWindow } from '@stencil/core/testing';
import { YooAlertComponent } from './alert';

jest.mock('../../../utils/anim');

describe('YooAlertComponent', () => {
    let window;
    beforeEach(() => {
        window = new TestWindow();
    });
    it('should build', () => {
        expect(new YooAlertComponent()).toBeTruthy();
    });

    describe('Snapshots', () => {
        it ('Should have text content', async () => {
            let element = await window.load({
               components: [YooAlertComponent],
               html: '<yoo-alert text="alert accent" class="accent toast"></yoo-alert>'
            });
            expect(element).toMatchSnapshot();
        });

        it ('Should have heading content', async () => {
            let element = await window.load({
               components: [YooAlertComponent],
               html: '<yoo-alert heading="alert accent" class="accent toast"></yoo-alert>'
            });
            expect(element).toMatchSnapshot();
        });

         it ('Should have icon content', async () => {
            let element = await window.load({
               components: [YooAlertComponent],
               html: '<yoo-alert icon="yo-check-tick" class="accent toast"></yoo-alert>'
            });
            expect(element).toMatchSnapshot();
        });

        it ('Should have closeable content', async () => {
            let element = await window.load({
               components: [YooAlertComponent],
               html: '<yoo-alert closeable="true" class="accent toast"></yoo-alert>'
            });
            expect(element).toMatchSnapshot();
        });

         it ('Should have heading and text content', async () => {
            let element = await window.load({
               components: [YooAlertComponent],
               html: '<yoo-alert heading="alert accent" text="alert text" class="accent toast"></yoo-alert>'
            });
            expect(element).toMatchSnapshot();
        });

          it ('Should have heading and text and closeable content', async () => {
            let element = await window.load({
               components: [YooAlertComponent],
               html: '<yoo-alert heading="alert accent" text="alert text" closeable="true" class="accent toast"></yoo-alert>'
            });
            expect(element).toMatchSnapshot();
        });

          it ('Should have heading and text and icon content', async () => {
            let element = await window.load({
               components: [YooAlertComponent],
               html: '<yoo-alert heading="alert accent" text="alert text" icon="yo-check-tick" class="accent toast"></yoo-alert>'
            });
            expect(element).toMatchSnapshot();
        });

          it ('Should have heading and text and icon and closeable content', async () => {
            let element = await window.load({
               components: [YooAlertComponent],
               html: '<yoo-alert heading="alert accent" text="alert text" icon="yo-check-tick" closeable="true" class="accent toast"></yoo-alert>'
            });
            expect(element).toMatchSnapshot();
        });

    });

    describe('Events', () => {

        it('should trigger click close event when dismiss', async () => {
            let element = await window.load({
                components: [YooAlertComponent],
                html: '<yoo-alert text="alert" class="accent toast" closeable="true"></yoo-alert>'
             });

            let closeButton: HTMLElement = element.querySelector('.close-container > .close');
            expect(element.querySelector('.container').classList).not.toContain('closed');
            let res;
            let emitItem = jest.fn(ev => {res = ev.detail; });
            window.document.addEventListener('alertClosed', emitItem);
            await closeButton.click();
            expect(emitItem).toHaveBeenCalled();
            expect(res).toEqual(true);
            await window.flush();
            expect(element.querySelector('.container').classList).toContain('closed');
        });

    });

});
