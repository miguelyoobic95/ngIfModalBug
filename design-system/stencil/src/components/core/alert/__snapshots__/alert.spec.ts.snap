// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooAlertComponent Snapshots Should have closeable content 1`] = `
<yoo-alert
  class="accent toast hydrated"
  closeable="true"
  data-yoo-alert-host=""
>
  
  <div
    class="container"
    data-yoo-alert=""
  >
    <div
      class="link"
      data-yoo-alert=""
    >
      
    </div>
    <div
      class="inner-container"
      data-yoo-alert=""
    >
      
      <div
        class="text-container"
        data-yoo-alert=""
      >
        
        <span
          class="value"
          data-yoo-alert=""
        >
          
        </span>
      </div>
    </div>
    <div
      class="close-container"
      data-yoo-alert=""
    >
      <span
        class="close"
        data-yoo-alert=""
      >
         
        <i
          class="yo-close"
          data-yoo-alert=""
        />
      </span>
    </div>
  </div>
</yoo-alert>
`;

exports[`YooAlertComponent Snapshots Should have heading and text and closeable content 1`] = `
<yoo-alert
  class="accent toast hydrated"
  closeable="true"
  data-yoo-alert-host=""
  heading="alert accent"
  text="alert text"
>
  
  <div
    class="container"
    data-yoo-alert=""
  >
    <div
      class="link"
      data-yoo-alert=""
    >
      
    </div>
    <div
      class="inner-container"
      data-yoo-alert=""
    >
      
      <div
        class="text-container"
        data-yoo-alert=""
      >
        <span
          class="heading"
          data-yoo-alert=""
        >
          alert accent
        </span>
        <span
          class="value"
          data-yoo-alert=""
        >
          alert text
        </span>
      </div>
    </div>
    <div
      class="close-container"
      data-yoo-alert=""
    >
      <span
        class="close"
        data-yoo-alert=""
      >
         
        <i
          class="yo-close"
          data-yoo-alert=""
        />
      </span>
    </div>
  </div>
</yoo-alert>
`;

exports[`YooAlertComponent Snapshots Should have heading and text and icon and closeable content 1`] = `
<yoo-alert
  class="accent toast hydrated"
  closeable="true"
  data-yoo-alert-host=""
  heading="alert accent"
  icon="yo-check-tick"
  text="alert text"
>
  
  <div
    class="container"
    data-yoo-alert=""
  >
    <div
      class="link"
      data-yoo-alert=""
    >
      
    </div>
    <div
      class="inner-container"
      data-yoo-alert=""
    >
      <span
        class="icon"
        data-yoo-alert=""
      >
        <i
          class="yo-check-tick"
          data-yoo-alert=""
        />
      </span>
      <div
        class="text-container"
        data-yoo-alert=""
      >
        <span
          class="heading"
          data-yoo-alert=""
        >
          alert accent
        </span>
        <span
          class="value"
          data-yoo-alert=""
        >
          alert text
        </span>
      </div>
    </div>
    <div
      class="close-container"
      data-yoo-alert=""
    >
      <span
        class="close"
        data-yoo-alert=""
      >
         
        <i
          class="yo-close"
          data-yoo-alert=""
        />
      </span>
    </div>
  </div>
</yoo-alert>
`;

exports[`YooAlertComponent Snapshots Should have heading and text and icon content 1`] = `
<yoo-alert
  class="accent toast hydrated"
  data-yoo-alert-host=""
  heading="alert accent"
  icon="yo-check-tick"
  text="alert text"
>
  
  <div
    class="container"
    data-yoo-alert=""
  >
    <div
      class="link"
      data-yoo-alert=""
    >
      
    </div>
    <div
      class="inner-container"
      data-yoo-alert=""
    >
      <span
        class="icon"
        data-yoo-alert=""
      >
        <i
          class="yo-check-tick"
          data-yoo-alert=""
        />
      </span>
      <div
        class="text-container"
        data-yoo-alert=""
      >
        <span
          class="heading"
          data-yoo-alert=""
        >
          alert accent
        </span>
        <span
          class="value"
          data-yoo-alert=""
        >
          alert text
        </span>
      </div>
    </div>
    <div
      class="close-container"
      data-yoo-alert=""
    >
      
    </div>
  </div>
</yoo-alert>
`;

exports[`YooAlertComponent Snapshots Should have heading and text content 1`] = `
<yoo-alert
  class="accent toast hydrated"
  data-yoo-alert-host=""
  heading="alert accent"
  text="alert text"
>
  
  <div
    class="container"
    data-yoo-alert=""
  >
    <div
      class="link"
      data-yoo-alert=""
    >
      
    </div>
    <div
      class="inner-container"
      data-yoo-alert=""
    >
      
      <div
        class="text-container"
        data-yoo-alert=""
      >
        <span
          class="heading"
          data-yoo-alert=""
        >
          alert accent
        </span>
        <span
          class="value"
          data-yoo-alert=""
        >
          alert text
        </span>
      </div>
    </div>
    <div
      class="close-container"
      data-yoo-alert=""
    >
      
    </div>
  </div>
</yoo-alert>
`;

exports[`YooAlertComponent Snapshots Should have heading content 1`] = `
<yoo-alert
  class="accent toast hydrated"
  data-yoo-alert-host=""
  heading="alert accent"
>
  
  <div
    class="container"
    data-yoo-alert=""
  >
    <div
      class="link"
      data-yoo-alert=""
    >
      
    </div>
    <div
      class="inner-container"
      data-yoo-alert=""
    >
      
      <div
        class="text-container"
        data-yoo-alert=""
      >
        <span
          class="heading"
          data-yoo-alert=""
        >
          alert accent
        </span>
        <span
          class="value"
          data-yoo-alert=""
        >
          
        </span>
      </div>
    </div>
    <div
      class="close-container"
      data-yoo-alert=""
    >
      
    </div>
  </div>
</yoo-alert>
`;

exports[`YooAlertComponent Snapshots Should have icon content 1`] = `
<yoo-alert
  class="accent toast hydrated"
  data-yoo-alert-host=""
  icon="yo-check-tick"
>
  
  <div
    class="container"
    data-yoo-alert=""
  >
    <div
      class="link"
      data-yoo-alert=""
    >
      
    </div>
    <div
      class="inner-container"
      data-yoo-alert=""
    >
      <span
        class="icon"
        data-yoo-alert=""
      >
        <i
          class="yo-check-tick"
          data-yoo-alert=""
        />
      </span>
      <div
        class="text-container"
        data-yoo-alert=""
      >
        
        <span
          class="value"
          data-yoo-alert=""
        >
          
        </span>
      </div>
    </div>
    <div
      class="close-container"
      data-yoo-alert=""
    >
      
    </div>
  </div>
</yoo-alert>
`;

exports[`YooAlertComponent Snapshots Should have text content 1`] = `
<yoo-alert
  class="accent toast hydrated"
  data-yoo-alert-host=""
  text="alert accent"
>
  
  <div
    class="container"
    data-yoo-alert=""
  >
    <div
      class="link"
      data-yoo-alert=""
    >
      
    </div>
    <div
      class="inner-container"
      data-yoo-alert=""
    >
      
      <div
        class="text-container"
        data-yoo-alert=""
      >
        
        <span
          class="value"
          data-yoo-alert=""
        >
          alert accent
        </span>
      </div>
    </div>
    <div
      class="close-container"
      data-yoo-alert=""
    >
      
    </div>
  </div>
</yoo-alert>
`;
